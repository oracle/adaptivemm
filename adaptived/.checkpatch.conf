# There is no Kernel tree available, run without it.
--no-tree

# Ignore the SPDX License tag, that's supposed to be the first line in
# Linux Kernel sources.
--ignore SPDX_LICENSE_TAG

# Follow the 100-character limit.
--max-line-length=100

# Kernel specific kstr* functions recommendation doesn't apply to us.
--ignore SSCANF_TO_KSTRTO

# Ignore Kernel str* function recommendations
--ignore STRCPY
--ignore STRNCPY

# Ignore the suggestion to add file changes (add, move, or delete) to the
# MAINTAINERS file
--ignore FILE_PATH_CHANGES
