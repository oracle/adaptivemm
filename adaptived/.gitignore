*~
*.css
*.la
*.lo
*.o
*.pc
*.swp
*.orig
*.gcno
*.gcda
*.html
*.info
*.log
*.trs
*-coverage/
.deps
.dirstamp
.libs
configure
Makefile
Makefile.in
cscope.*
tags
gtest
/autom4te.cache
/aclocal.m4
/build-aux
/config.*
/configure
/configure.h*
/libtool
/m4
/stamp-h1
/cov-int
/src/adaptived
