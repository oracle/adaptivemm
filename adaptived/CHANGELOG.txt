adaptived: Releases
===============================================================================
https://github.com/oracle/adaptivemm

* Version 1.0.1 - February 24, 2025
- Fix bug where only two or fewer rules could be loaded
- Fix duration bug in PSI cause
- Fix memory leak reported by coverity

* Version 1.0.0 - November 8, 2024
- Initial release
