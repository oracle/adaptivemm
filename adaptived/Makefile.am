ACLOCAL_AMFLAGS = -I m4
DIST_SUBDIRS = dist doc include src tests
SUBDIRS = ${DIST_SUBDIRS}

EXTRA_DIST = \
	CONTRIBUTING.md \
	LICENSE.txt \
	README.md \
	SECURITY.md \
	THIRD_PARTY_LICENSES.txt

help:
	@echo "adaptived build system"
	@echo " make targets:"
	@echo "  (none):           build the library"
	@echo "  check:            build the library and run tests"
	@echo "  check-build:      build the library and all tests"

check-build: all
	${MAKE} ${AM_MAKEFLAGS} -C src check-build
	${MAKE} ${AM_MAKEFLAGS} -C tests check-build
