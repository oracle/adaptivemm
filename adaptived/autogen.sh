#!/bin/bash
#

git submodule update --init --recursive

# configure googletest
pushd googletest/googletest
git checkout release-1.8.0
cmake -DBUILD_SHARED_LIBS=ON .
make
popd

test -d m4 || mkdir m4
autoreconf -fi
rm -rf autom4te.cache
