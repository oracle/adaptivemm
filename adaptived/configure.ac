dnl # Copyright (c) 2024, Oracle and/or its affiliates.
dnl # DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
dnl #
dnl # This code is free software; you can redistribute it and/or modify it
dnl # under the terms of the GNU General Public License version 2 only, as
dnl # published by the Free Software Foundation.
dnl #
dnl # This code is distributed in the hope that it will be useful, but WITHOUT
dnl # ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
dnl # FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
dnl # version 2 for more details (a copy is included in the LICENSE file that
dnl # accompanied this code).
dnl #
dnl # You should have received a copy of the GNU General Public License version
dnl # 2 along with this work; if not, write to the Free Software Foundation,
dnl # Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
dnl #
dnl # Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
dnl # or visit www.oracle.com if you need additional information or have any
dnl # questions.

AC_INIT([adaptived],[0.0.0])

dnl #
dnl # autoconf configuration
dnl #
AC_CONFIG_AUX_DIR([build-aux])
AC_CONFIG_HEADERS([configure.h])
AC_CONFIG_MACRO_DIR([m4])
AM_INIT_AUTOMAKE(-Wall foreign subdir-objects)

m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
m4_ifdef([AM_PROG_AR], [AM_PROG_AR])

dnl #
dnl # build flags
dnl #
AM_CPPFLAGS="-I\${top_srcdir}/include -I\${top_builddir}/include"
AM_CFLAGS="-Wall"
AM_LDFLAGS="-Wl,-z -Wl,relro"
AC_SUBST([AM_CPPFLAGS])
AC_SUBST([AM_CFLAGS])
AC_SUBST([AM_LDFLAGS])

dnl #
dnl # version information
dnl #
VERSION_MAJOR=$(echo ${VERSION} | cut -d'.' -f 1)
VERSION_MINOR=$(echo ${VERSION} | cut -d'.' -f 2)
VERSION_MICRO=$(echo ${VERSION} | cut -d'.' -f 3)
AC_SUBST([VERSION_MAJOR])
AC_SUBST([VERSION_MINOR])
AC_SUBST([VERSION_MICRO])

dnl #
dnl # Determine which C compiler to use
dnl #
AC_PROG_CXX
AC_PROG_CC
AM_PROG_CC_C_O

dnl #
dnl # Dependencies
dnl #
AC_CHECK_LIB([json-c], [json_object_object_get_ex], [],
	     [AC_MSG_ERROR([please install json-c])])

AC_CHECK_HEADERS([json-c/json.h], [],
		 [AC_MSG_ERROR([please install the json-c development package])])

LT_INIT

dnl #
dnl # Enable/Disable code coverage
dnl #
AX_CODE_COVERAGE

dnl #
dnl # makefiles
dnl #
AC_CONFIG_FILES([
	Makefile
	dist/Makefile
	doc/Makefile
	doc/examples/Makefile
	include/Makefile
	src/Makefile
	tests/Makefile
	tests/ftests/Makefile
	tests/gunit/Makefile
])
AC_CONFIG_FILES([dist/adaptived.spec:dist/adaptived.spec.in])

dnl #
dnl # generate the configure and makefiles
dnl #
AC_OUTPUT
