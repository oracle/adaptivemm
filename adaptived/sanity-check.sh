#!/bin/bash
# Copyright (c) 2024, Oracle and/or its affiliates.
# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
#
# This code is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License version 2 only, as
# published by the Free Software Foundation.
#
# This code is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
# version 2 for more details (a copy is included in the LICENSE file that
# accompanied this code).
#
# You should have received a copy of the GNU General Public License version
# 2 along with this work; if not, write to the Free Software Foundation,
# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
#
# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
# or visit www.oracle.com if you need additional information or have any
# questions.

# clean up from a previous run
make clean
find . -name "*.gcno" -exec sudo rm '{}' \;
find . -name "*.gcda" -exec sudo rm '{}' \;

./autogen.sh
RET=$?

if [[ $RET -ne 0 ]]; then
	echo "./autogen failed: " $RET
	exit $RET
fi

CFLAGS="$CFLAGS -g -O0 -Werror" ./configure
RET=$?

if [[ $RET -ne 0 ]]; then
	echo "./configure failed: " $RET
	exit $RET
fi

make distcheck
RET=$?

if [[ $RET -ne 0 ]]; then
	echo "make distcheck failed: " $RET
	exit $RET
fi

CFLAGS="$CFLAGS -g -O0 -Werror" ./configure --enable-code-coverage
RET=$?

if [[ $RET -ne 0 ]]; then
	echo "./configure w/ code coverage failed: " $RET
	exit $RET
fi

# I'm not fond of running all of the tests as root, but the sd-bus tests
# need administrative privileges.  Merging the non-sudo and sudo code
# coverage files is difficult.  Let's avoid that.
sudo make check
RET=$?

if [[ $RET -ne 0 ]]; then
	echo "make check failed: " $RET
	exit $RET
fi

gcovr -s --filter src/
exit 0
