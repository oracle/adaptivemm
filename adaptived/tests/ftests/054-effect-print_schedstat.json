{
	"rules": [
		{
			"name": "Run the print_schedstat effect",
			"causes": [
				{
					"name": "periodic",
					"description": "period is in milliseconds",
					"args": {
						"period": 1000
					}
				}
			],
			"effects": [
				{
					"name": "print_schedstat",
					"args": {
						"file": "stdout",
						"schedstat_file": "054-effect-print_schedstat.schedstat"
					}
				}
			]
		}
	]
}
