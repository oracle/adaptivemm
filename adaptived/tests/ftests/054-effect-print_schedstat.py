#!/usr/bin/env python3
# Copyright (c) 2024, Oracle and/or its affiliates.
# DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
#
# This code is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License version 2 only, as
# published by the Free Software Foundation.
#
# This code is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
# version 2 for more details (a copy is included in the LICENSE file that
# accompanied this code).
#
# You should have received a copy of the GNU General Public License version
# 2 along with this work; if not, write to the Free Software Foundation,
# Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
#
# Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
# or visit www.oracle.com if you need additional information or have any
# questions.

#
# Test the print_schedstat effect
#
#

import filecmp
import adaptived
import consts
import ftests
import sys
import os

CONFIG = '054-effect-print_schedstat.json'
OUT_FILE = '054-effect-print_schedstat.out'
EXPECTED_RET = 62
MAX_LOOPS = 1

def prereqs(config):
    result = consts.TEST_PASSED
    cause = None

    return result, cause


def setup(config):
    pass


def test(config):
    result = consts.TEST_PASSED
    cause = None

    out = adaptived.adaptived(config=CONFIG, max_loops=MAX_LOOPS, expected_ret=EXPECTED_RET)

    with open(OUT_FILE, 'w') as ofile:
        ofile.write(out)
        ofile.write('\n')

    match = filecmp.cmp(OUT_FILE, '054-effect-print_schedstat.expected', shallow=False)
    if not match:
        result = consts.TEST_FAILED
        cause = 'Expected out file differs from received out file'

    return result, cause


def teardown(config):
    pass


def main(config):
    [result, cause] = prereqs(config)
    if result != consts.TEST_PASSED:
        return [result, cause]

    try:
        setup(config)
        [result, cause] = test(config)
    finally:
        teardown(config)

    try:
        os.remove(OUT_FILE)
    except FileNotFoundError:
        pass

    return [result, cause]


if __name__ == '__main__':
    config = ftests.parse_args()
    # this test was invoked directly.  run only it
    config.args.num = int(os.path.basename(__file__).split('-')[0])
    sys.exit(ftests.main(config))

# vim: set et ts=4 sw=4:
