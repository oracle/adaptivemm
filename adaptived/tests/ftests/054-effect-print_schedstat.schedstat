version 15
timestamp 6037440866
cpu0 157155 0 36291220 23981035 12755074 4114542 44364538015765 408312389517 441756650
domain0 00003 5420675 5420365 154 716 162 0 0 5420358 11787 11787 0 0 0 0 0 11787 523358 520591 1299 4022 1468 0 0 514835 6 0 6 0 0 0 0 0 0 115353 5607 0
domain1 fffff 1105444 972868 132248 154547 185075 3 1662 924839 27 27 0 0 0 0 0 1 507624 223325 273192 356636 11107 340 7550 215607 186389 1642 184746 0 0 0 0 0 0 8525179 1321566 0
cpu1 51590 0 8259189 5433206 2747145 938576 11207917802972 141685096013 103647542
domain0 00003 1544758 1543909 150 1091 704 0 0 1543687 17413 17413 0 0 0 0 0 17413 164090 159353 2762 5455 1975 0 0 155274 0 0 0 0 0 0 0 0 0 104138 8541 0
domain1 fffff 544651 544206 376 605 22635 0 54 19813 672 672 0 0 0 0 0 0 152014 76597 71755 109538 3662 0 1959 74515 22902 291 22611 0 0 0 0 0 0 1704431 247643 0
cpu2 182337 0 29958095 15072700 14632546 4574573 55621022317917 614418458603 543922228
domain0 0000c 4164793 4164266 408 602 185 0 0 4163915 57196 57196 0 0 0 0 0 57196 302287 299631 1376 2831 1280 0 0 293734 0 0 0 0 0 0 0 0 0 131278 5197 0
domain1 fffff 927548 771569 155481 181528 152976 76 2767 696288 680 680 0 0 0 0 0 14 285639 155702 122219 169485 7718 39 3431 152116 153037 576 152461 0 0 0 0 0 0 9926695 1507072 0
cpu3 70007 0 6706410 3542357 3537787 1191323 9380929502228 179875987674 87980628
domain0 0000c 1614847 1614178 464 772 247 0 0 1614089 15480 15479 1 85 0 0 0 15479 167603 165327 1032 2360 1244 0 5 161586 0 0 0 0 0 0 0 0 0 141453 17980 0
domain1 fffff 555944 553980 1935 2974 22806 0 164 55610 798 798 0 0 0 0 0 0 155185 87833 63980 94259 3372 0 1402 86311 22959 192 22767 0 0 0 0 0 0 2205011 442345 0
cpu4 168765 0 32645671 16357054 16542117 5184700 141296917922429 940856152641 579154830
domain0 00030 4541079 4540009 911 1683 253 0 0 4539870 61647 61647 0 0 0 0 0 61647 276903 272290 3273 4814 1340 0 0 266689 1 0 1 0 0 0 0 0 0 304287 32174 0
domain1 fffff 1051946 738151 313236 368529 323117 165 6922 662462 501 501 0 0 0 0 0 12 261722 98225 157751 269157 5746 73 3705 94326 323517 981 322537 0 0 0 0 0 0 11053130 1753081 0
cpu5 55464 0 10419640 5405070 5086164 1854442 39606971096353 364760871050 167192239
domain0 00030 1777007 1774458 2152 2641 456 0 0 1774425 57742 57742 0 0 0 0 0 57742 126093 117239 7022 9144 1832 0 3 113438 1 0 1 0 0 0 0 0 0 294994 61802 0
domain1 fffff 615590 612701 2823 32213 24047 1 104 101838 877 877 0 0 0 0 0 0 113796 57758 53270 95981 2768 1 1116 56528 24280 294 23986 0 0 0 0 0 0 2936728 471198 0
cpu6 188252 0 37527797 18701854 19064821 6471638 150263969164205 1141062127199 620948990
domain0 000c0 8765015 8764073 716 2078 230 0 0 8763983 10151 10151 0 0 0 0 0 10151 419039 414124 3301 5069 1614 0 1 408794 4 0 4 0 0 0 0 0 0 371982 35088 0
domain1 fffff 1332944 1030730 301919 324202 372565 28 5006 982948 334 334 0 0 0 0 0 4 403063 188688 208387 264177 5988 35 7127 181355 373504 1235 372269 0 0 0 0 0 0 12221201 1954858 0
cpu7 44136 0 10973191 5689105 5359404 1970480 35428861757599 239566832889 146159730
domain0 000c0 1740483 1733568 5889 426676 3595 0 6 1733460 56527 56524 2 109 1 0 0 56524 114763 102138 10683 202330 1942 0 26 98463 2555 11 2544 0 0 0 0 0 0 359188 82640 0
domain1 fffff 602077 600326 1719 1872 20439 0 98 110116 1518 1518 0 0 0 0 0 0 105171 56117 46671 73187 2383 0 1415 54603 20789 381 20408 0 0 0 0 0 0 3029736 530788 0
cpu8 200593 0 38740338 19380963 18542876 6028775 90898330087562 874245132339 698130398
domain0 00300 5347970 5347832 165 348 130 0 0 5347779 39632 39632 0 0 0 0 0 39632 395795 394168 562 1711 1065 0 1 388089 0 0 0 0 0 0 0 0 0 101803 5635 0
domain1 fffff 1079996 933888 145665 169190 158180 24 4569 862339 79 79 0 0 0 0 0 0 380867 221458 150621 204262 8788 25 5532 215746 158451 709 157742 0 0 0 0 0 0 12412298 1963857 0
cpu9 48247 0 4195722 2327859 1778718 521737 8274479521101 113502649232 75332259
domain0 00300 1319168 1318074 692 1156 423 0 2 1318025 12726 12726 0 0 0 0 0 12726 90631 87256 2138 3441 1237 0 0 83870 0 0 0 0 0 0 0 0 0 88276 10874 0
domain1 fffff 481036 478538 2429 3784 9331 1 332 83859 616 616 0 0 0 0 0 0 78396 39122 37286 53633 1988 0 833 38180 9342 82 9260 0 0 0 0 0 0 1168705 145850 0
cpu10 191412 0 36443808 18248951 17429906 5164748 80243206483910 580925360545 618582967
domain0 00c00 5026352 5026428 131 224 92 0 0 5026386 43314 43313 1 137 0 0 0 43313 373651 372371 361 1318 919 0 0 366481 1 0 1 0 0 0 0 0 0 93248 4182 0
domain1 fffff 1046029 901193 144373 167707 160833 23 4473 836423 111 111 0 0 0 0 0 0 358132 202579 147521 199348 8032 27 5621 196770 161005 643 160362 0 0 0 0 0 0 12171910 1804363 0
cpu11 18992 0 3448706 1976289 1398746 352069 8786080279149 175257818796 80148397
domain0 00c00 1236201 1235144 682 6260 428 0 2 1235071 9664 9662 1 77 1 0 0 9662 77284 74754 1306 3131 1224 0 0 71134 39 0 39 0 0 0 0 0 0 74122 12505 0
domain1 fffff 454991 452794 2133 2364 6548 0 262 69917 571 571 0 0 0 0 0 0 63614 28196 33842 47213 1576 0 709 27385 6538 55 6483 0 0 0 0 0 0 972555 114318 0
cpu12 208992 0 25150778 12576410 11675875 3053513 56525551701231 633142656790 432527730
domain0 0f000 2054563 2053308 1197 2157 254 0 2 2053308 24699 24697 1 14 1 0 0 24697 375575 363770 7466 12288 4339 0 6 358504 0 0 0 0 0 0 0 0 0 4117116 1088577 0
domain1 fffff 819736 799979 19192 22310 22783 1 299079 406807 424 424 0 0 0 0 0 6 361363 275572 78171 101800 7620 4 210500 64934 22340 150 22190 0 0 0 0 0 0 4505246 295481 0
cpu13 214096 0 22374964 11213667 10325249 3261347 48712060061604 676010987422 394579860
domain0 0f000 1936028 1935214 809 2225 201 0 1 1935212 22374 22371 2 236 1 0 0 22371 327654 316948 6512 11140 4194 0 4 312169 0 0 0 0 0 0 0 0 0 3928909 919183 0
domain1 fffff 798189 795400 2723 2875 14667 0 21576 32213 337 337 0 0 0 0 0 0 312009 236915 69176 82505 5918 0 169540 67238 14688 105 14583 0 0 0 0 0 0 3134993 221215 0
cpu14 181503 0 17913726 9037082 8142512 2056463 41149436189137 434815380686 317996924
domain0 0f000 1602847 1601845 960 2165 213 0 0 1601844 32347 32346 1 79 0 0 0 32346 261835 251377 6430 10824 4028 1 7 247005 0 0 0 0 0 0 0 0 0 3669442 836840 0
domain1 fffff 698397 697424 966 1022 12089 0 1704 6265 1399 1399 0 0 0 0 0 0 248044 182791 60470 71260 4783 0 131882 50796 12132 68 12064 0 0 0 0 0 0 2416607 187394 0
cpu15 166298 0 14643890 7430225 6483698 1534461 35161497374103 363849759068 264569816
domain0 0f000 1424798 1423809 901 1381 222 0 1 1423806 32769 32765 3 295 1 0 0 32765 222998 212628 6461 10720 3909 0 7 208690 0 0 0 0 0 0 0 0 0 3049359 766700 0
domain1 fffff 646074 645534 529 564 10871 0 193 1885 1707 1707 0 0 0 0 0 0 210082 150878 55094 67090 4110 0 106088 44690 10916 66 10850 0 0 0 0 0 0 1899878 161962 0
cpu16 81977 0 7845426 4048767 4241499 1092928 22222085829523 459990108253 214768632
domain0 f0000 1204696 1203809 821 1828 151 0 2 1203802 13986 13986 0 0 0 0 0 13986 234286 226991 4704 7683 2591 0 6 220974 0 0 0 0 0 0 0 0 0 1314550 317521 0
domain1 fffff 565693 535694 29366 31846 23196 12 272153 185897 1370 1370 0 0 0 0 0 39 218868 141323 72095 85675 5450 15 66862 74401 22610 62 22548 0 0 0 0 0 0 1834021 161067 0
cpu17 73037 0 6951961 3602977 2854413 724788 22476418603452 419542718964 223823838
domain0 f0000 1123640 1122737 809 1596 160 0 7 1122729 15179 15178 1 64 0 0 0 15178 245949 238181 5196 8134 2572 0 10 232430 0 0 0 0 0 0 0 0 0 982759 216449 0
domain1 fffff 533203 531113 2020 3108 13800 0 5766 10099 1792 1792 0 0 0 0 0 0 231292 161660 65489 74967 4143 0 80894 80710 13783 57 13726 0 0 0 0 0 0 1146866 102040 0
cpu18 66467 0 5983810 3143045 2414440 575298 21862181641620 411934728524 213636087
domain0 f0000 1070764 1069871 799 1053 143 0 2 1069867 14323 14322 1 21 0 0 0 14322 212973 205720 4884 7623 2369 0 10 200169 0 0 0 0 0 0 0 0 0 867790 201726 0
domain1 fffff 510985 510393 580 697 10851 0 379 1910 1715 1715 0 0 0 0 0 0 198301 135666 59182 69175 3453 0 66594 69011 10875 40 10835 0 0 0 0 0 0 971352 88157 0
cpu19 46528 0 5212369 2784631 2090481 476180 21416282687499 421265610974 207328014
domain0 f0000 1033777 1032948 748 1141 131 0 5 1032942 13338 13338 0 0 0 0 0 13338 179283 172642 4412 6983 2229 0 14 167383 0 0 0 0 0 0 0 0 0 737770 177219 0
domain1 fffff 492932 492659 271 297 9749 0 47 457 1673 1673 0 0 0 0 0 0 165225 107608 54562 63304 3055 0 51885 55657 9785 41 9744 0 0 0 0 0 0 876531 79034 0
