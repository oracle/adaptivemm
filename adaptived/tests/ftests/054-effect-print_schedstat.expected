Timestamp (jiffies/ticks): 6037440866:
CPU0:
	Number of wakeups from this CPU: 12755074
	Number of wakeups to this CPU:  4114542
	Total task run time (nanoseconds):  44364538015765
	Total task wait time (nanoseconds):  408312389517
	Number of timeslices on this CPU:  441756650
Domain0:
	Number of remote wakeups: 115353
	Number of affine wakeups: 5607
Domain1:
	Number of remote wakeups: 8525179
	Number of affine wakeups: 1321566
CPU1:
	Number of wakeups from this CPU: 2747145
	Number of wakeups to this CPU:  938576
	Total task run time (nanoseconds):  11207917802972
	Total task wait time (nanoseconds):  141685096013
	Number of timeslices on this CPU:  103647542
Domain0:
	Number of remote wakeups: 104138
	Number of affine wakeups: 8541
Domain1:
	Number of remote wakeups: 1704431
	Number of affine wakeups: 247643
CPU2:
	Number of wakeups from this CPU: 14632546
	Number of wakeups to this CPU:  4574573
	Total task run time (nanoseconds):  55621022317917
	Total task wait time (nanoseconds):  614418458603
	Number of timeslices on this CPU:  543922228
Domain0:
	Number of remote wakeups: 131278
	Number of affine wakeups: 5197
Domain1:
	Number of remote wakeups: 9926695
	Number of affine wakeups: 1507072
CPU3:
	Number of wakeups from this CPU: 3537787
	Number of wakeups to this CPU:  1191323
	Total task run time (nanoseconds):  9380929502228
	Total task wait time (nanoseconds):  179875987674
	Number of timeslices on this CPU:  87980628
Domain0:
	Number of remote wakeups: 141453
	Number of affine wakeups: 17980
Domain1:
	Number of remote wakeups: 2205011
	Number of affine wakeups: 442345
CPU4:
	Number of wakeups from this CPU: 16542117
	Number of wakeups to this CPU:  5184700
	Total task run time (nanoseconds):  141296917922429
	Total task wait time (nanoseconds):  940856152641
	Number of timeslices on this CPU:  579154830
Domain0:
	Number of remote wakeups: 304287
	Number of affine wakeups: 32174
Domain1:
	Number of remote wakeups: 11053130
	Number of affine wakeups: 1753081
CPU5:
	Number of wakeups from this CPU: 5086164
	Number of wakeups to this CPU:  1854442
	Total task run time (nanoseconds):  39606971096353
	Total task wait time (nanoseconds):  364760871050
	Number of timeslices on this CPU:  167192239
Domain0:
	Number of remote wakeups: 294994
	Number of affine wakeups: 61802
Domain1:
	Number of remote wakeups: 2936728
	Number of affine wakeups: 471198
CPU6:
	Number of wakeups from this CPU: 19064821
	Number of wakeups to this CPU:  6471638
	Total task run time (nanoseconds):  150263969164205
	Total task wait time (nanoseconds):  1141062127199
	Number of timeslices on this CPU:  620948990
Domain0:
	Number of remote wakeups: 371982
	Number of affine wakeups: 35088
Domain1:
	Number of remote wakeups: 12221201
	Number of affine wakeups: 1954858
CPU7:
	Number of wakeups from this CPU: 5359404
	Number of wakeups to this CPU:  1970480
	Total task run time (nanoseconds):  35428861757599
	Total task wait time (nanoseconds):  239566832889
	Number of timeslices on this CPU:  146159730
Domain0:
	Number of remote wakeups: 359188
	Number of affine wakeups: 82640
Domain1:
	Number of remote wakeups: 3029736
	Number of affine wakeups: 530788
CPU8:
	Number of wakeups from this CPU: 18542876
	Number of wakeups to this CPU:  6028775
	Total task run time (nanoseconds):  90898330087562
	Total task wait time (nanoseconds):  874245132339
	Number of timeslices on this CPU:  698130398
Domain0:
	Number of remote wakeups: 101803
	Number of affine wakeups: 5635
Domain1:
	Number of remote wakeups: 12412298
	Number of affine wakeups: 1963857
CPU9:
	Number of wakeups from this CPU: 1778718
	Number of wakeups to this CPU:  521737
	Total task run time (nanoseconds):  8274479521101
	Total task wait time (nanoseconds):  113502649232
	Number of timeslices on this CPU:  75332259
Domain0:
	Number of remote wakeups: 88276
	Number of affine wakeups: 10874
Domain1:
	Number of remote wakeups: 1168705
	Number of affine wakeups: 145850
CPU10:
	Number of wakeups from this CPU: 17429906
	Number of wakeups to this CPU:  5164748
	Total task run time (nanoseconds):  80243206483910
	Total task wait time (nanoseconds):  580925360545
	Number of timeslices on this CPU:  618582967
Domain0:
	Number of remote wakeups: 93248
	Number of affine wakeups: 4182
Domain1:
	Number of remote wakeups: 12171910
	Number of affine wakeups: 1804363
CPU11:
	Number of wakeups from this CPU: 1398746
	Number of wakeups to this CPU:  352069
	Total task run time (nanoseconds):  8786080279149
	Total task wait time (nanoseconds):  175257818796
	Number of timeslices on this CPU:  80148397
Domain0:
	Number of remote wakeups: 74122
	Number of affine wakeups: 12505
Domain1:
	Number of remote wakeups: 972555
	Number of affine wakeups: 114318
CPU12:
	Number of wakeups from this CPU: 11675875
	Number of wakeups to this CPU:  3053513
	Total task run time (nanoseconds):  56525551701231
	Total task wait time (nanoseconds):  633142656790
	Number of timeslices on this CPU:  432527730
Domain0:
	Number of remote wakeups: 4117116
	Number of affine wakeups: 1088577
Domain1:
	Number of remote wakeups: 4505246
	Number of affine wakeups: 295481
CPU13:
	Number of wakeups from this CPU: 10325249
	Number of wakeups to this CPU:  3261347
	Total task run time (nanoseconds):  48712060061604
	Total task wait time (nanoseconds):  676010987422
	Number of timeslices on this CPU:  394579860
Domain0:
	Number of remote wakeups: 3928909
	Number of affine wakeups: 919183
Domain1:
	Number of remote wakeups: 3134993
	Number of affine wakeups: 221215
CPU14:
	Number of wakeups from this CPU: 8142512
	Number of wakeups to this CPU:  2056463
	Total task run time (nanoseconds):  41149436189137
	Total task wait time (nanoseconds):  434815380686
	Number of timeslices on this CPU:  317996924
Domain0:
	Number of remote wakeups: 3669442
	Number of affine wakeups: 836840
Domain1:
	Number of remote wakeups: 2416607
	Number of affine wakeups: 187394
CPU15:
	Number of wakeups from this CPU: 6483698
	Number of wakeups to this CPU:  1534461
	Total task run time (nanoseconds):  35161497374103
	Total task wait time (nanoseconds):  363849759068
	Number of timeslices on this CPU:  264569816
Domain0:
	Number of remote wakeups: 3049359
	Number of affine wakeups: 766700
Domain1:
	Number of remote wakeups: 1899878
	Number of affine wakeups: 161962
CPU16:
	Number of wakeups from this CPU: 4241499
	Number of wakeups to this CPU:  1092928
	Total task run time (nanoseconds):  22222085829523
	Total task wait time (nanoseconds):  459990108253
	Number of timeslices on this CPU:  214768632
Domain0:
	Number of remote wakeups: 1314550
	Number of affine wakeups: 317521
Domain1:
	Number of remote wakeups: 1834021
	Number of affine wakeups: 161067
CPU17:
	Number of wakeups from this CPU: 2854413
	Number of wakeups to this CPU:  724788
	Total task run time (nanoseconds):  22476418603452
	Total task wait time (nanoseconds):  419542718964
	Number of timeslices on this CPU:  223823838
Domain0:
	Number of remote wakeups: 982759
	Number of affine wakeups: 216449
Domain1:
	Number of remote wakeups: 1146866
	Number of affine wakeups: 102040
CPU18:
	Number of wakeups from this CPU: 2414440
	Number of wakeups to this CPU:  575298
	Total task run time (nanoseconds):  21862181641620
	Total task wait time (nanoseconds):  411934728524
	Number of timeslices on this CPU:  213636087
Domain0:
	Number of remote wakeups: 867790
	Number of affine wakeups: 201726
Domain1:
	Number of remote wakeups: 971352
	Number of affine wakeups: 88157
CPU19:
	Number of wakeups from this CPU: 2090481
	Number of wakeups to this CPU:  476180
	Total task run time (nanoseconds):  21416282687499
	Total task wait time (nanoseconds):  421265610974
	Number of timeslices on this CPU:  207328014
Domain0:
	Number of remote wakeups: 737770
	Number of affine wakeups: 177219
Domain1:
	Number of remote wakeups: 876531
	Number of affine wakeups: 79034
